hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/nssummary/DiskUsageSubCommand.java

```diff
@@ -0,0 +1,214 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.ozone.admin.nssummary;
+
+import com.google.gson.internal.LinkedTreeMap;
+import org.apache.commons.io.FileUtils;
+import org.apache.hadoop.hdds.cli.HddsVersionProvider;
+import org.apache.hadoop.ozone.shell.ListOptions;
+import picocli.CommandLine;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.concurrent.Callable;
+
+import static org.apache.hadoop.ozone.OzoneConsts.OM_KEY_PREFIX;
+import static org.apache.hadoop.ozone.admin.nssummary.NSSummaryCLIUtils.getResponseMap;
+import static org.apache.hadoop.ozone.admin.nssummary.NSSummaryCLIUtils.makeHttpCall;
+import static org.apache.hadoop.ozone.admin.nssummary.NSSummaryCLIUtils.parseInputPath;
+import static org.apache.hadoop.ozone.admin.nssummary.NSSummaryCLIUtils.printEmptyPathRequest;
+import static org.apache.hadoop.ozone.admin.nssummary.NSSummaryCLIUtils.printFSOReminder;
+import static org.apache.hadoop.ozone.admin.nssummary.NSSummaryCLIUtils.printKVSeparator;
+import static org.apache.hadoop.ozone.admin.nssummary.NSSummaryCLIUtils.printNewLines;
+import static org.apache.hadoop.ozone.admin.nssummary.NSSummaryCLIUtils.printPathNotFound;
+import static org.apache.hadoop.ozone.admin.nssummary.NSSummaryCLIUtils.printSpaces;
+import static org.apache.hadoop.ozone.admin.nssummary.NSSummaryCLIUtils.printWithUnderline;
+
+/**
+ * Disk Usage Subcommand.
+ */
+@CommandLine.Command(
+    name = "du",
+    description = "Get disk usage for a path request.",
+    mixinStandardHelpOptions = true,
+    versionProvider = HddsVersionProvider.class
+)
+
+public class DiskUsageSubCommand implements Callable {
+  @CommandLine.ParentCommand
+  private NSSummaryAdmin parent;
+
+  @CommandLine.Parameters(index = "0", arity = "0..1",
+      description = "Non-empty path request without any protocol prefix.")
+  private String path;
+
+  @CommandLine.Option(names = {"-f", "--file"},
+      description = "List direct files as a sub path.")
+  private boolean listFiles;
+
+  @CommandLine.Option(names = {"-r", "--replica"},
+      description = "Show disk usage with replication.")
+  private boolean withReplica;
+
+  @CommandLine.Option(names = {"-n", "--no-header"},
+      description = "Show DU without the header for current path.")
+  private boolean noHeader;
+
+  @CommandLine.Mixin
+  private ListOptions listOptions;
+
+  private static final String ENDPOINT = "/api/v1/namespace/du";
+
+  // For text alignment
+  private static final String SIZE_HEADER = "Size";
+  private static final String DU_HEADER = "Disk Usage";
+  private static final String PATH_HEADER = "Path Name";
+  private static final int SIZE_INDENT = 2;
+  private static final int DU_INDENT = 12;
+  private static final int PATH_INDENT = 27;
+
+  private StringBuffer url = new StringBuffer();
+
+  @Override
+  public Void call() throws Exception {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 18. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.